core/server/errors.js

```diff
@@ -0,0 +1,154 @@
+var _ = require('lodash'),
+    util = require('util');
+
+function GhostError(options) {
+    options = options || {};
+
+    if (_.isString(options)) {
+        throw new Error('Please instantiate Errors with the option pattern. e.g. new errors.GhostError({message: ...})');
+    }
+
+    Error.call(this);
+    Error.captureStackTrace(this, GhostError);
+
+    /**
+     * defaults
+     */
+    this.statusCode = 500;
+    this.errorType = 'InternalServerError';
+    this.level = 'normal';
```

Comment:
TODO Note: I'd like to add the usage of an individual ID to errors, as we have in ignition